app/src/main/java/com/example/drawingApp/ColorPickerUtil.kt

```diff
@@ -0,0 +1,201 @@
+         * on null don't do anything with it, don't set rgb and hsv to defaults.
+         */
+        onSubmission: (Color?) -> Unit
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.color_picker_bottom_sheet, null)
+        val submit = view.findViewById<TextView>(R.id.dialogSubmit)
+        val cancel = view.findViewById<TextView>(R.id.dialogCancel)
+        val hueView = view.findViewById<ForceEditText>(R.id.hslH)
+        val saturationView = view.findViewById<ForceEditText>(R.id.hslS)
+        val valueView = view.findViewById<ForceEditText>(R.id.hslL)
+        val rgbRedView = view.findViewById<ForceEditText>(R.id.rgbR)
+        val rgbGreenView = view.findViewById<ForceEditText>(R.id.rgbG)
+        val rgbBlueView = view.findViewById<ForceEditText>(R.id.rgbB)
+        val colorHexView = view.findViewById<ForceEditText>(R.id.colorHex)
+        val slider = view.findViewById<ColorSlider>(R.id.slider)
+        val square = view.findViewById<GradientSquare>(R.id.gradientSquare)
+        val activeColorBox = view.findViewById<TextView>(R.id.colorOne)
+        val previousColorBox = view.findViewById<TextView>(R.id.colorTwo)
+        val previousColor = onColorTextForceUpdate().previousColor
+        previousColorBox.setBackgroundColor(previousColor.toArgb())
+        //used to set the color of the active box
+        val onSettingActiveBox: (FloatArray) -> Unit = { hsv ->
+            val intColor = Color.HSVToColor(hsv)
+            val red = Color.red(intColor)
+            val green = Color.green(intColor)
+            val blue = Color.blue(intColor)
+
+            hueView.forceText(hsv[0].toString())
+            saturationView.forceText((hsv[1] * 100).toString())
+            valueView.forceText((hsv[2] * 100).toString())
+            rgbRedView.forceText(red.toString())
+            rgbGreenView.forceText(green.toString())
+            rgbBlueView.forceText(blue.toString())
+            colorHexView.forceText(Integer.toHexString(Color.rgb(red, green, blue)).drop(2))
+
+            square.hue = hsv[0]
+            square.sat = hsv[1]
+            square.valu = hsv[2]
+            square.invalidate()
+
+            slider.sliderSetHue(hsv[0])
+
+            activeColorBox.setBackgroundColor(intColor)
+        }
+        //called whenever text needs to be updated
+        val updateForceViews = {
+            val pack = onColorTextForceUpdate()
+            onSettingActiveBox(floatArrayOf(pack.hue, pack.saturation, pack.value))
+        }
+        previousColorBox.setOnClickListener {
+            val hsv = FloatArray(3)
+            Color.colorToHSV(previousColor.toArgb(), hsv)
+            //used to set the color of the active box to the previous color
+            onSettingActiveBox(hsv)
+        }
+        submit.setOnClickListener {
+            onSubmission(
+                Color.HSVToColor(
```

Comment:
Use your fancy Hsv object with the method you're gonna make for this

Reply:
implemented